spec/index.md

```diff
@@ -3702,6 +3738,77 @@ The functions `query_as_update` and `system_task_as_update` turns a query functi
 
 Note that by construction, a query function will either trap or return with a response; it will never send calls, and it will never change the state of the canister.
 
+#### Spontaneous request rejection {#request-rejection}
+
+The system can reject a request at any point in time, e.g., because it is overloaded. The following specification is an overapproximation of the rejection behavior. In particular, the system is guaranteed not to emit `DESTINATION_INVALID` in some cases, e.g., if the callee exists, and if the caller successfully calls the caller once, subsequent calls will not fail with `DESTINATION_INVALID`.
+
+Condition:
+```html
+S.messages = Older_messages · CallMessage CM · Younger_messages
+(CM.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+reject_code ∈ { SYS_FATAL, SYS_TRANSIENT, DESTINATION_INVALID }
+∃reject_msg: True
+```
+
+State after:
+```html
+S.messages =
+    Older_messages ·
+    ResponseMessage {
+        origin = CM.origin;
+        response = Reject (reject_code, reject_msg);
+        refunded_cycles = CM.transferred_cycles;
+        deadline = CM.deadline;
+    } ·
+    Younger_messages
+```
+
+#### Call expiry {#call-expiry}
+
+These transitions expire calls with timeouts. To account for SYS_UNKNOWN being issued early (e.g., due to high system load), we ignore the caller time in these transitions. We define two variants of the transition, one that expires messages, and one that expires calls that are in progress (i.e., have open downstream call contexts).
+
+The first transition defines the expiry of messages.
+
+```html
+S.messages = Older_messages · M · Younger_messages
+M = CallMessage _ ∨ M = ResponseMessage _
+M.origin = FromCanister _
+M.deadline ∉ { NoDeadline, Expired }
+∃reject_msg: True
+```
+
+State after
+```html
+S.messages = Older_messages · M { deadline = Expired } · Younger_messages ·
+    ResponseMessage {
+        origin = CM.origin;
+        response = Reject (SYS_UNKNOWN, reject_msg);
+        refunded_cycles = 0;
+        deadline = NoDeadline;
+    }
+```
+
+The next transition defines the expiry of calls that are being processed by the callee.
```

Comment:
Wouldn't we have two reject messages with SYS_UNKNOWN at the end?

Reply:
As discussed in the meeting last week, the call life cycle ensures that only one `CallMessage`, `CallContext` or a `ResponseMessage` exists for a given call at a given point in time. With response timeouts, this still holds as long as we ignore those values whose deadline is `Expired` (even if they get converted to `ResponseMessage` eventually, the response will not be delivered but eventually garbage collected).

Reply:
> the call life cycle ensures that only one CallMessage, CallContext or a ResponseMessage exists for a given call at a given point in time

There can a CallContext and ResponseMessage at the same time.